pool.go

```diff
@@ -61,3 +74,31 @@ func (p *ObjectPool) Put(obj interface{}) {
 	default:
 	}
 }
+
+// Init initializes the string slice pool.
+func (p *StringSlicePool) Init(alloc func() []string) {
+	p.alloc = alloc
+
+	for i := 0; i < cap(p.values); i++ {
+		p.values <- p.alloc()
+	}
+}
+
+// Get gets a string slice from the pool.
+func (p *StringSlicePool) Get() []string {
+	var v []string
+	select {
+	case v = <-p.values:
+	default:
+		v = p.alloc()
+	}
+	return v
+}
+
+// Put releases a string slice back to the pool.
+func (p *StringSlicePool) Put(obj []string) {
```

Comment:
Interestingly Put(obj interface{}) with a []string escapes to heap but not with *bytes.Buffer